src/com/lftechnology/java/training/sanish/application/model/Model.java

```diff
@@ -0,0 +1,55 @@
+package com.lftechnology.java.training.sanish.application.model;
+
+import java.sql.ResultSet;
+
+/**
+ * Db model interface
+ *
+ * @param <M> {@link Model}
+ * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+ */
+public interface Model <M extends Model>{
+    /**
+     * Return table name
+     *
+     * @return {@link String}
+     * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+     */
+    public String getTable();
+
+    /**
+     * Return value of attribute
+     *
+     * @param key {@link String} attribute name
+     * @param <V>
+     * @return
+     * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+     */
+    public <V> V getAttribute(String key);
+
+    /**
+     * Set value of attribute
+     *
+     * @param key {@link String} attribute name
+     * @param <V>
+     * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+     */
+    public <V> void setAttribute(String key, V value);
+
+    /**
+     * Check other model equal with current model
+     *
+     * @param model {@link Model}
+     * @return {@link Boolean}
+     * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+     */
+    public boolean equals(M model);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either override Object.equals(Object), or totally rename the method to prevent any confusion. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1201)